src/main/java/org/ow2/proactive/workflow_catalog/rest/service/WorkflowRevisionService.java

```diff
@@ -277,13 +277,29 @@ private QueryExpressionContext createJpaQueryExpression(
         WorkflowMetadata workflowMetadata =
                 new WorkflowMetadata(workflowRevision);
 
-        workflowMetadata.add(createLink(bucketId, workflowId, revisionId, workflowRevision));
+        workflowMetadata.add(createLink(bucketId, workflowId, workflowRevision));
 
         return ResponseEntity.ok(workflowMetadata);
     }
 
-    public Link createLink(Long bucketId, Long workflowId, Optional<Long> revisionId,
-            WorkflowRevision workflowRevision) {
+    public ResponseEntity<?> delete(Long bucketId, Long workflowId, Optional<Long> revisionId) {
+        Workflow workflow = findWorkflow(workflowId);
+        WorkflowRevision workflowRevision = null;
+        if (revisionId.isPresent() && workflow.getRevisions().size() > 1) {
+            // TODO: remove the current revision R and point the workflow to revision R-1
```

Comment:
You are missing a TODO (I go through the commits in a series, can be that it is fixed in a later commit which I didn't see yet).